flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/RowDataToProtoSerializer.java

```diff
@@ -0,0 +1,256 @@
+                                element.getTimestamp(fieldNumber, 3).getMillisecond(),
+                                false,
+                                "Timestamp(millis)");
+                    } else {
+                        return AvroToProtoSerializer.AvroSchemaHandler.convertTimestamp(
+                                element.getTimestamp(fieldNumber, 6).getMillisecond(),
+                                true,
+                                "Timestamp(micros)");
+                    }
+                case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                    // microseconds since epoch
+                    if (((LocalZonedTimestampType) fieldType).getPrecision() == 3) {
+                        return AvroToProtoSerializer.AvroSchemaHandler.convertDateTime(
+                                element.getTimestamp(fieldNumber, 3).getMillisecond(), false);
+                    } else {
+                        return AvroToProtoSerializer.AvroSchemaHandler.convertDateTime(
+                                element.getTimestamp(fieldNumber, 6).getMillisecond(), true);
+                    }
+                case ARRAY:
+                    LogicalType arrayElementType = getArrayElementType(fieldType);
+                    ArrayData.ElementGetter elementGetter =
+                            ArrayData.createElementGetter(arrayElementType);
+                    List<Object> ans =
+                            Stream.iterate(0, pos -> pos + 1)
+                                    .limit(element.getArray(fieldNumber).size())
+                                    .map(
+                                            pos ->
+                                                    convertArrayElement(
+                                                            elementGetter,
+                                                            element,
+                                                            fieldNumber,
+                                                            pos,
+                                                            arrayElementType,
+                                                            fieldDescriptor))
+                                    .collect(Collectors.toList());
+                    return ans;
+                case INTERVAL_YEAR_MONTH:
+                case INTERVAL_DAY_TIME:
+                case MAP:
+                case MULTISET:
+                case NULL:
+                case SYMBOL:
+                case RAW:
+                case DISTINCT_TYPE:
+                case STRUCTURED_TYPE:
+                case TIMESTAMP_WITH_TIME_ZONE:
+                    // Since it is only a logical type and not supported in flink SQL
+                    // (https://issues.apache.org/jira/browse/FLINK-20869)
+                case UNRESOLVED:
+                    throw new UnsupportedOperationException("Not supported yet!");
+            }
+        } catch (Exception e) {
```

Comment:
This is a bad practice to catch the base exception and throw a RuntimeException. Please catch specific exceptions.

Reply:
Fixed.